buildSrc/src/main/kotlin/PublishPlugin.kt

```diff
@@ -39,21 +46,22 @@ abstract class PublishPlugin : Plugin<Project> {
             IosPlugin.TARGET_NAME_X64 to mppTarget.ios,
             IosPlugin.TARGET_NAME_SIM to mppTarget.ios
         )
-        target.extensions.findByType(KotlinMultiplatformExtension::class)?.apply {
-            targets.configureEach { configureMavenPublicationTasksForEnvironment(this, taskConfigurationMap) }
-        }
     }
 
-    private fun configureMavenPublicationTasksForEnvironment(
-        kotlinTarget: KotlinTarget,
+    private fun setupLocalPublishing(
+        target: Project,
         taskConfigurationMap: Map<String, Boolean>
     ) {
-        kotlinTarget.mavenPublication {
-            kotlinTarget.project.tasks.withType(AbstractPublishToMaven::class).configureEach {
-                onlyIf {
-                    taskConfigurationMap[publication.name] == true
-                }
+        target.project.tasks.withType(AbstractPublishToMaven::class).configureEach {
+            val configuration = publication?.name ?: run {
+                // Android Plugin does not set publication property after creation of task
+                logger.warn("Unable to configure task $name in place, using hacks instead")
+                val configuration = taskConfigurationMap.keys
+                    .find { name.contains(it, ignoreCase = true) }
+                logger.warn("Found $configuration for $name")
```

Comment:
What if not found?

Reply:
Publishing will be disabled for this configuration.